devito/targets/basic.py

```diff
@@ -0,0 +1,153 @@
+
+
+def iet_lower(iet, mode='advanced', options=None):
+    """
+    Specialize an Iteration/Expression tree (IET) for a certain Target. This
+    consists of:
+
+        * Loop-level transformations to generate Target-optimized code;
+        * Shared-memory and distributed-memory parallelism;
+        * Symbol and data management.
+
+    Parameters
+    ----------
+    iet : Node
+        The IET to be lowered.
+    mode : str, optional
+        The DLE transformation mode.
+        - ``noop``: Do nothing.
+        - ``advanced``: flush denormals, vectorization, loop blocking, OpenMP-related
+                        optimizations (collapsing, nested parallelism, ...), MPI-related
+                        optimizations (aggregation of communications, reshuffling of
+                        communications, ...).
+    options : dict, optional
+        - ``openmp``: Enable/disable OpenMP. Defaults to `configuration['openmp']`.
+        - ``mpi``: Enable/disable MPI. Defaults to `configuration['mpi']`.
+        - ``blockinner``: Enable/disable blocking of innermost loops. By default,
+                          this is disabled to maximize SIMD vectorization. Pass True
+                          to override this heuristic.
+        - ``blocklevels``: Levels of blocking for hierarchical tiling (blocks,
+                           sub-blocks, sub-sub-blocks, ...). Different Platforms have
+                           different default values.
+        - ``language``: To generate code using a low-level language other than the
+                        default one. Currently this option is ignored, but in the
+                        future we expect to have multiple alternatives for the same
+                        platform. For example, if the platform in question is NVIDIAX,
+                        one may want to generate OpenMP or OpenACC offloading code --
+                        ``language=openmp`` and ``language=openacc`` respectively) --
+                        rather than the default ``language=cuda``.
+    """
+    assert isinstance(iet, Node)
+
+    # What is the target platform for which the optimizations are applied?
+    platform = configuration['platform']
+
+    # Default options
+    params = {}
+    params['blockinner'] = configuration['dle-options'].get('blockinner', False)
+    params['blocklevels'] = configuration['dle-options'].get('blocklevels', None)
+    params['openmp'] = configuration['openmp']
+    params['mpi'] = configuration['mpi']
+
+    # Parse input options (potentially replacing defaults)
+    for k, v in (options or {}).items():
+        if k not in params:
+            warning("Illegal DLE option '%s'" % k)
+        else:
+            params[k] = v
+
+    # Force OpenMP/MPI if parallelism was requested, even though mode is 'noop'
+    if mode is None:
+        mode = 'noop'
+    elif mode == 'noop':
+        mode = tuple(i for i in ['mpi', 'openmp'] if params[i]) or 'noop'
+
+    # Fetch the requested rewriter
```

Comment:
Should the 'rewriter' terminology be changed after the Rewriter  classes are not there anymore?
I mean no classes from DLE anymore as a term. Just a thought...

Reply:
yes, you're right

Reply:
Dropping this in a subsequent PR. Will leave it for now, but it was a fair point